docs/side_quests/ide_features.md

```diff
@@ -0,0 +1,654 @@
+
+### Takeaway
+
+You can manage complex multi-file projects using quick file navigation, split editors, and project-wide search to work efficiently across workflows and modules.
+
+### What's next?
+
+Learn how code formatting and maintenance features keep your workflows organized and readable.
+
+---
+
+## 6. Code Formatting and Maintenance
+
+Good code formatting isn't just about aesthetics - it makes complex workflows easier to understand and maintain.
+
+### 6.1. Automatic Formatting in Action
+
+Open `basic_workflow.nf` and deliberately mess up the formatting:
+
+- Remove some indentation
+- Add extra spaces in random places
+- Break some lines awkwardly
+
+Now press **Shift+Alt+F** (or **Shift+Option+F**) to auto-format:
+
+VS Code immediately:
+
+- Fixes indentation to show process structure clearly
+- Aligns similar elements consistently
+- Removes unnecessary whitespace
+- Maintains readable line breaks
+
+### 6.2. Code Organization Features
+
+#### Quick Commenting
+
+Select a block of code in your workflow and press **Ctrl+/** (or **Cmd+/**) to comment it out:
+
+```groovy
+// workflow {
+//     ch_input = Channel.fromPath(params.input)
+//         .splitCsv(header: true)
+//         .map { row -> [row.sample_id, file(row.fastq_path)] }
+//
+//     FASTQC(ch_input)
+// }
+```
+
+This is perfect for:
+
+- Temporarily disabling parts of workflows during development
+- Adding explanatory comments to complex channel operations
+- Documenting workflow sections
+
+Use **Ctrl+/** (or **Cmd+/**) again to uncomment the code.
+
+#### Code Folding for Overview
+
+In `complex_workflow.nf`, notice the small arrows next to process definitions. Click them to fold (collapse) processes:
+
+![Code folding](img/code_folding.png)
+
+This gives you a high-level overview of your workflow structure without getting lost in implementation details.
+
+#### Bracket Matching
+
+Place your cursor next to any `{` or `}` bracket and VS Code highlights the matching bracket. Use **Ctrl+Shift+\\** (or **Cmd+Shift+\\**) to jump between matching brackets.
+
+This is crucial for:
+
+- Understanding process boundaries
+- Finding missing or extra brackets
+- Navigating nested workflow structures
+
+### Takeaway
+
+You can maintain clean, readable code using automatic formatting, commenting features, code folding, and bracket matching to organize complex workflows.
+
+### What's next?
+
+Learn how VS Code integrates with your broader development workflow beyond just editing code.
+
+---
+
+## 7. Development Workflow Integration
+
+VS Code integrates well with your development workflow beyond just editing code.
+
+### 7.1. Version Control Integration
+
+If your project is a git repository (as this is), VS Code shows:
+
+- Modified files with colored indicators
+- Git status in the status bar
+- Inline diff views
+- Commit and push capabilities
+
+Open the Source Control panel (`Ctrl+Shift+G` or `Cmd+Shift+G`) to see git changes and stage commits directly in the editor.
```

Comment:
another one that doesn't work in codespaces

Reply:
I also had previously said no to opening it as git repository, when doing `code .` on the directory as we would normally not need this in training.

Reply:
I would find a screenshot here helpful, there are a bunch of options to select from on the side, (if the short cut doesn't work or does something unexpected)

Reply:
Agreed, thanks, clarifications and image added